x-pack/plugins/connectors_networking/server/docs/poc.md

```diff
@@ -0,0 +1,345 @@
+Would one space want to verify a server, and another not?
+
+I guess we have to think about the future.  What if we want to add something here
+that _should_ be space specific.  It would certainly be some new sort of data,
+so would probably require a new saved object, which _could_ be space specific.
+
+### CNO's are unique by protocol : hostname : port
+
+For now anyway, it seems possible to get by with having CNO objects be unique
+just by protocol / hostname / port - without considering any path, query string
+parameters, user/pass, etc. TLS certificates only go to this granularity as
+well. If needed, in the future, we could extend this with path's or other
+factors.
+
+We'll canonicalize the url, by only using the protocol, hostname, and port from
+provided urls, and add the default port for the protocol if not provided in the
+url.
+
+This breaks for the only generic option, `timeout`.  Presumably you should be
+able to set a specific timeout for a specific url pattern that could include
+a path.  For example, you might use the default timeout for most requests,
+but then a certain API like `/api/digits-of-pi` might need a longer timeout,
+so it would be nice to constrain the longer timeout to just that URL.  Perhaps
+that's another good reason to remove the `timeout` option.  We could re-visit
+allowing path elements (and query string params? headers?) later.
+
+### CNO's are "matched" to URLs, not specifically assigned
+
+Rather than have to assign a CNO to a specific connector, we're taking
+advantage of the fact that TLS options are specific to the hostname and port,
+and only applicable to https and smtp.  This allows us to create the CNO's
+independent of the connectors - they could be created BEFORE the connectors
+are created.  And once created, and new connectors using the same hostname
+and port will pick up the CNO without any direct association.
+
+This works out great, in practice, but am wondering if I missed some gotchas
+with it.
+
+In fact, having written that bit above about creating CNO's "early", these CNO's
+are perhaps good fodder for Kibana config - a way of pre-registering them, where
+we could (potentially) have the customers provide PEMs as file names (for
+on-prem; clould would need to support PEM's as strings of the file contents (7
+bit ascii anyway!)).
+
+In fact, having written that, if we could do this, we could perhaps ship this
+without any management UI for now, and only allow CNO's via config.  We 
+wouldn't need the saved objects at all.  
+
+:wondering:
+
+### CNO ids will be canonical URLs
+
+Kinda.  We don't want `/` chars in there, because we want to use "ids" as 
+path parameters for REST APIs, so we'll create keys with protocol / hostname /
+port, separated by `:` chars.  Eg, URL `https://elastic.co` becomes key/id
+`https:elastic.co:443`.
+
+I know generally we use UUIDs as saved object ids, and especially for encrypted
```

Comment:
[UUIDv5 generates deterministic UUIDs](https://en.wikipedia.org/wiki/Universally_unique_identifier#Versions_3_and_5_(namespace_name-based)), so it'd still be possible to use a UUID-based unique identifier, while still ensuring that two CNOs can't share a canonical URL.

We're using this within the Saved Objects migrator as part of the multi-namespace conversion process:
https://github.com/elastic/kibana/blob/dc475c9cf2a7f42c6b7df921a6dc8613765c80d7/src/core/server/saved_objects/migrations/core/document_migrator.ts#L847-L856